File: languages/go/entity/integrity/anchor.go
package integrity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type Anchor struct {
	Id         int64
	BlockRoots []string
	Networks   []AnchorNetwork
	Root       string
	Status     string
}

func NewAnchorFromProto(a *proto.Anchor) Anchor {
	if a == nil {
		return Anchor{}
	}

	return Anchor{
		Id:         a.Id,
		BlockRoots: a.BlockRoots,
		Networks:   MapAnchorNetworksFromProto(a.Networks),
		Root:       a.Root,
		Status:     a.Status,
	}
}

File: languages/js/src/entity/integrity/anchor.ts
import * as proto from "../../bridge/proto/integrity_entities";
import { AnchorNetwork } from "./anchor_network";

/**
 * Represents information about an anchor.
 */
export class Anchor {
  id: number;
  blockRoots: string[];
  networks: AnchorNetwork[];
  root: string;
  status: string;

  /**
   * Constructs an Anchor object with the specified parameters.
   * @param id 
   * @param blockRoots 
   * @param networks 
   * @param root 
   * @param status 
   */
  constructor(
    id: number,
    blockRoots: string[],
    networks: AnchorNetwork[],
    root: string,
    status: string
  ) {
    this.id = id;
    this.blockRoots = blockRoots;
    this.networks = networks;
    this.root = root;
    this.status = status;
  }

  static fromProto(a: proto.Anchor): Anchor {
    return new Anchor(
      a.id,
      a.blockRoots,
      a.networks.map(n => AnchorNetwork.fromProto(n)),
      a.root,
      a.status
    );
  }
}
